test/grouping.test.tsx

```diff
@@ -0,0 +1,197 @@
+import { StrictMode, useState } from 'react';
+import { groupBy as rowGrouper } from 'lodash';
+import type { Column } from '../src';
+import DataGrid, { SelectColumn } from '../src';
+import { render, screen, within } from '@testing-library/react';
+import { getGrid, queryGrid, getRows, queryTreeGrid, getTreeGrid } from './utils';
+import userEvent from '@testing-library/user-event';
+
+interface Row {
+  id: number;
+  country: string;
+  year: number;
+}
+
+const columns: readonly Column<Row>[] = [
+  SelectColumn,
+  {
+    key: 'country',
+    name: 'Country'
+  },
+  {
+    key: 'year',
+    name: 'Year'
+  },
+  {
+    key: 'sport',
+    name: 'Sport'
+  }
+];
+
+const rows: Row[] = [
+  {
+    id: 1,
+    country: 'USA',
+    year: 2020
+  },
+  {
+    id: 2,
+    country: 'USA',
+    year: 2021
+  },
+  {
+    id: 3,
+    country: 'Canada',
+    year: 2021
+  },
+  {
+    id: 4,
+    country: 'Canada',
+    year: 2022
+  }
+];
+
+function rowKeyGetter(row: Row) {
+  return row.id;
+}
+
+function TestGrid({ groupBy }: { groupBy?: string[] }) {
+  const [selectedRows, setSelectedRows] = useState<ReadonlySet<number>>(() => new Set());
+  const [expandedGroupIds, setExpandedGroupIds] = useState<ReadonlySet<unknown>>(
+    () => new Set<unknown>([])
+  );
+
+  return (
+    <DataGrid
+      columns={columns}
+      rows={rows}
+      rowKeyGetter={rowKeyGetter}
+      groupBy={groupBy}
+      rowGrouper={rowGrouper}
+      selectedRows={selectedRows}
+      onSelectedRowsChange={setSelectedRows}
+      expandedGroupIds={expandedGroupIds}
+      onExpandedGroupIdsChange={setExpandedGroupIds}
+    />
+  );
+}
+
+function setup(groupBy?: string[]) {
+  return render(
```

Comment:
Doesn't really matter, you can ignore this if you want
```suggestion
  render(
```